src/backends/pytorch/libtorch_backend.cc

```diff
@@ -460,324 +546,215 @@ LibTorchBackend::Context::GetOutputTensor(
-    }
-  }
-#endif  // TRITON_ENABLE_STATS
+  INFER_STATS_DECL_TIMESTAMP(compute_output_start_ns);
 
-  // verify output indices are valid with number of outputs after execution
-  for (const auto& output : base->Config().output()) {
-    int op_index = output_index_map_[output.name()];
-    int max_index = outputs_.size() - 1;
+  // Verify output indices are valid with number of outputs after execution
+  int max_index = outputs_.size() - 1;
+  for (const auto& name : required_outputs) {
+    int op_index = output_index_map_[name];
     if ((op_index < 0) || (op_index > max_index)) {
-      return Status(
+      Status status = Status(
           Status::Code::INVALID_ARG,
-          "The output " + output.name() +
+          "The output " + name +
               " in the model configuration refers to an output index which "
               "doesn't exist. This model has " +
               std::to_string(max_index + 1) + " outputs");
-    }
-  }
 
-  // Prepare set of Outputs requested for
-  std::set<std::string> required_outputs;
-  for (auto& payload : *payloads) {
-    const auto& irequest = payload.request_;
-    for (const auto& pr : irequest->RequestedOutputs()) {
-      required_outputs.insert(pr.first);
+      FAIL_ALL_AND_RETURN_IF_ERROR(
+          requests, responses, metric_reporter_.get(), status,
+          "error creating LibTorch output tensor");
     }
   }
 
-  // Ensure outputs have the expected size and copy it to the payload responses.
-  std::vector<OutputInfo> outputs;
-  cuda_copy = false;
-  for (const auto& name : required_outputs) {
-    outputs.emplace_back();
-    int op_index = output_index_map_[name];
-    const ModelOutput* output_config;
-    RETURN_IF_ERROR(base->GetOutput(name, &output_config));
-
-    const DataType dtype = output_config->data_type();
-
-    // If a reshape is provided for the output then use that when
-    // validating that the model matches what is expected.
-    const DimsList& output_dims = (output_config->has_reshape())
-                                      ? output_config->reshape().shape()
-                                      : output_config->dims();
-
-    // Checked at initialization time to make sure that STRING is not
-    // being used for an output, so can just assume fixed-sized here.
-    RETURN_IF_ERROR(ReadFixedSizedOutputTensor(
-        &outputs_, name, op_index, dtype,
-        GetDataTypeByteSize(output_config->data_type()), total_batch_size,
-        output_dims, payloads, &outputs.back(), &cuda_copy));
-  }
+  // Create the response tensors and copy the appropriate tensor data
+  // into each.
+  FAIL_ALL_AND_RETURN_IF_ERROR(
+      requests, responses, metric_reporter_.get(),
+      ReadOutputTensors(
+          base, total_batch_size, requests, &responses, &outputs_,
+          &output_index_map_),
+      "error sending LibTorch response");
 
 #ifdef TRITON_ENABLE_GPU
   if (cuda_copy) {
     cudaStreamSynchronize(stream_);
```

Comment:
You don't need to synchronize here, already done in ReadOutputTensors

Reply:
Removed